packages/storage-plus/src/map.rs

```diff
@@ -635,6 +638,77 @@ mod test {
         assert_eq!(all, vec![(50, data3)]);
     }
 
+    #[test]
+    #[cfg(feature = "iterator")]
+    fn range_signed_integer_key_migration() {
+        let mut store = MockStorage::new();
+
+        // save and load three keys with the old format
+        let data = Data {
+            name: "John".to_string(),
+            age: 32,
+        };
+        SIGNED_ID_OLD
+            .save(&mut store, IntKeyOld::<i32>::from(-1234), &data)
+            .unwrap();
+
+        let data2 = Data {
+            name: "Jim".to_string(),
+            age: 44,
+        };
+        SIGNED_ID_OLD
+            .save(&mut store, IntKeyOld::<i32>::from(-56), &data2)
+            .unwrap();
+
+        let data3 = Data {
+            name: "Jules".to_string(),
+            age: 55,
+        };
+        SIGNED_ID_OLD
+            .save(&mut store, IntKeyOld::<i32>::from(50), &data3)
+            .unwrap();
+
+        // obtain all current keys
+        let current = SIGNED_ID_OLD
+            .range(&store, None, None, Order::Ascending)
+            .collect::<StdResult<Vec<_>>>()
+            .unwrap();
+        // confirm wrong current order
+        assert_eq!(
+            current,
+            vec![
+                (50, data3.clone()),
+                (-1234, data.clone()),
+                (-56, data2.clone())
+            ]
+        );
+
+        // remove old entries
+        for (k, _) in current.iter() {
+            SIGNED_ID_OLD.remove(&mut store, IntKeyOld::<i32>::from(*k));
```

Comment:
`k.into()` doesn't work here?

I thought `.remove()` is strongly typed arg, which makes this easier to write

Reply:
It does. Changed.